x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/lib/extract_mappings_definition.ts

```diff
@@ -32,6 +37,29 @@ const isMappingDefinition = (obj: GenericObject): boolean => {
   return isConfigurationValid && isPropertiesValid && isDynamicTemplatesValid;
 };
 
+const getMappingsDefinitionWithType = (mappings: GenericObject): MappingsWithType[] => {
+  if (isMappingDefinition(mappings)) {
+    // No need to go any further
+    return [{ mappings }];
+  }
+
+  // At this point there must be one or more type mappings
+  const typedMappings = Object.entries(mappings).reduce(
+    (acc: Array<{ type: string; mappings: GenericObject }>, [type, value]) => {
+      if (isMappingDefinition(value)) {
+        acc.push({ type, mappings: value as GenericObject });
+      }
+      return acc;
+    },
+    []
+  );
+
+  return typedMappings;
+};
+
+export const doesMappingsHasType = (mappings: GenericObject = {}): boolean =>
```

Comment:
Nit: I think the name `doMappingsHaveType` or even `isTypedMappings` would be more grammatically correct.

Reply:
For sure 😊